src/module/hotKeysModule.js

```diff
@@ -34,6 +34,10 @@ var HotKeysModule = class HotKeysModule {
         this.actionIdToNameMap = new Map();
         this.actionNameToActionMap = new Map();
 
+        this.cycleThroughtWorkspaces = getSettings('layouts').get_boolean(
```

Comment:
this.cycleThroughtWorkspaces is never updated so the shell need to be restarted every time the option is changed. To prevent either connect to the option changes to update this variable or even better read the settings value directly when needed (You can create a getter for this if you want like in the MsThemeManager)